packer/sagebio/sagebio-base-ubuntu-bionic-LATEST/playbook.yaml

```diff
@@ -17,13 +17,19 @@
           - "curl"
         state: present
 
-    # awscli is on aws linux2 distro by default
-    - name: install awscli
+    # awscli and amazon-ssm-agent are on aws linux2 by default
+    - name: Install awscli and amazon-ssm-agent if available in Snappy
+      snappable_distros: Ubuntu # a comma-separated list of distributions with snapd
       snap:
         name:
           - aws-cli
+          - amazon-ssm-agent
         classic: yes
-      when: ansible_distribution == "Ubuntu"
+      when: ansible_distribution in '{{ snappable_distros }}'
+
+    - name: Install amazon-ssm-agent if Centos
```

Comment:
what's the point of having this special task for amazon-ssm-agent?  centos supports snap repos as well.  https://snapcraft.io/install/snapd/centos

Reply:
afaik Centos does not come with snapd... so we would also need to install snapd separately. The special task is so I can follow the AWS install guide for amazon-ssm-agent on different platforms. They support snaps only on Ubuntu, and support direct download of rpm for Centos, RedHat, etc.

Reply:
hmm,  i believe ansible provides snapd functionality with the `snap:`?  and there is a amazon-ssm-agent package for centos (https://snapcraft.io/install/amazon-ssm-agent/centos) so i'm not sure why it wouldn't work.  you should test it though.

Reply:
I confirmed that ansible does not guarantee snapd is installed as prerequisite to snap